filelog.go

```diff
@@ -345,6 +369,108 @@ func (w *FileLogWriter) archiveFiles(dir string) error {
 	return nil
 }
 
+// Compress a file after it has been rotated.
+// plainFile - name of the rotated, uncompressed file
+// compressedInprogressFilename - name of a temporary file to hold compressed data
+// method - type of compression to use
+func (w *FileLogWriter) compressFile(plainFilename, compressedInprogressFilename, method string) bool {
```

Comment:
nit: `method` could be a specific type that aliases `string` (or `int`, if you like) with two constant values for `gzip` and `zip`.